### ISSUE TITLE
no-extra-parens handles some MemberExpressions incorrectly

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#9156]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 8.4.0
* **npm Version:** 5.3.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
rules:
  no-extra-parens: error
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
(foo()).bar;

new (foo.bar().Baz);
```

**What did you expect to happen?**

I expected an error to be reported on line 1, and no error on line 3.

**What actually happened? Please include the actual, raw output from ESLint.**

An error was reported on line 3, but no error was reported on line 1.

edit: added a false positive case

**Comment by jackyho112** [comment]
Mind if I give this a go? Will start working on this right away.

**Comment by not-an-aardvark** [comment]
Sure, feel free. Thanks for contributing!

**Comment by jackyho112** [comment]
@not-an-aardvark 

I have been familiarizing myself with the library and investigating the issue. I think I have a few ideas on how to fix them. I do want to confirm the logic with you though. The first line should have a warning because there should be no paren around a call expression? The third line should not have a warning because it is ok to have parens around a member expression with a new expression?

**Comment by platinumazure** [comment]
I think extra parens around a new expression should only be allowed if there are intervening parentheses.

```js
new (Foo.Bar.Baz);  // unnecessary parens
new (Foo.Bar().Baz);  // necessary parens

**Comment by jackyho112** [comment]
I will look into it more to find the necessary or unnecessary aspects of these parens.
